lib/audit-ci.js

```diff
@@ -47,16 +53,63 @@ const { argv } = yargs
   })
   .help('help');
 
+function mapVulnerabilityLevelInput(config) {
+  if (config.l) {
+    return { low: true, moderate: true, high: true, critical: true };
+  }
+  if (config.m) {
+    return { low: false, moderate: true, high: true, critical: true };
+  }
+  if (config.h) {
+    return { low: false, moderate: false, high: true, critical: true };
+  }
+  if (config.c) {
+    return { low: false, moderate: false, high: false, critical: true };
+  }
+  return { low: false, moderate: false, high: false, critical: false };
+}
+
+argv.levels = mapVulnerabilityLevelInput(argv);
+
+/**
+ * @param {'auto' | 'npm' | 'yarn'} pmArg the package manager (including the `auto` option)
+ * @returns {'npm' | 'yarn'} the non-`auto` package manager
+ */
+function getPackageManagerType(pmArg) {
+  switch (pmArg) {
+    case 'npm':
+      return 'npm';
+    case 'yarn':
+      return 'yarn';
+    case 'auto': {
+      const packageLockExists = fs.existsSync('package-lock.json');
+      if (packageLockExists) return 'npm';
+      const shrinkwrapExists = fs.existsSync('npm-shrinkwrap.json');
+      if (shrinkwrapExists) return 'npm';
+      const yarnLockExists = fs.existsSync('yarn.lock');
+      if (yarnLockExists) return 'yarn';
+      throw Error(
+        'Cannot establish package-manager type, missing package-lock.json and yarn.lock.'
+      );
+    }
+    default:
+      throw Error(`Unexpected package manager argument: ${pmArg}`);
+  }
+}
+
+const pm = getPackageManagerType(argv.p);
+
+const auditor =
+  pm === 'npm' ? require('./npm-auditer') : require('./yarn-auditer');
+
 auditor
   .audit(argv)
   .then(() => {
-    console.log('\x1b[32m%s\x1b[0m', 'Passed NPM security audit.');
+    console.log('\x1b[32m%s\x1b[0m', `Passed ${pm} security audit.`);
   })
   .catch(err => {
-    if (err) {
-      const message = err.message || 'An unknown error occurred.';
-      console.error('\x1b[31m%s\x1b[0m', message);
-      console.error('\x1b[31m%s\x1b[0m', 'Exiting...');
-      process.exitCode = 1;
-    }
+    const message = err.message || err;
```

Comment:
I don't love this approach. Since we mostly `reject(Error(...))`, the default is to use `err.message`. However, uncaught exceptions are often returned as a `string` (one example of this currently is not having a `package-lock.json` or `npm-shrinkwrap.json` when running with `npm`.